votrfront/js/sorting.js

```diff
@@ -73,3 +74,184 @@ export function sortTable(items, columns, query, queryKey) {
+                                order[0] == 'd' + index ? 'desc' : '')}>
+        {label}
+      </th>
+    )}
+  </tr>;
+
+  return [items, header];
+};
+
+function toggleInfo(index) {
+  const opened = JSON.parse(LocalSettings.get("openedRows")) || {};
+  opened[index] = !opened[index];
+  LocalSettings.set("openedRows", JSON.stringify(opened));
+}
+
+function expandAll(items) {
+  const opened = items.reduce((acc, curr, idx) => ({...acc, [idx]: true}), {});
+  LocalSettings.set("openedRows", JSON.stringify(opened));
+}
+
+// TODO: better names
+function collapseAll() {
+  LocalSettings.set("openedRows", JSON.stringify({}));
+}
+
+function isOpened(index) {
+  return (JSON.parse(LocalSettings.get("openedRows")) || {})[index];
+}
+
+export const SortableTable = ({
+  items,
+  columns,
+  queryKey,
+  withButtons,
+  footer,
+  message
+}) =>
+  queryConsumer(query => {
+    // TODO change to sortTable
+    const [sortedItems, header] = sortTableCopyForObjectCol(
+      items,
+      columns,
+      query,
+      queryKey
+    );
+
+    const className =
+      "table table-condensed table-bordered table-striped table-hover" +
+      (withButtons ? " with-buttons-table" : "");
+
+    const notExpandable = columns.reduce(
+      (acc, col) =>
+        acc.filter(item => !(col.hiddenClass && col.hiddenClass.includes(item))),
+      ["hidden-xs", "hidden-sm", "hidden-md", "hidden-lg"]
+    );
+
+    const rows = [];
+
+    sortedItems.forEach((item, index) => {
+      rows.push(
+        <tr key={index} onClick={() => toggleInfo(index)}>
+          {columns.map(
+            ({
+              label,
+              prop,
+              process,
+              hiddenClass = [],
+              cell,
+              colProps,
+              expansionMark
+            }) => (
+              <td
+                key={label}
+                className={hiddenClass.join(" ")}
+                {...(colProps instanceof Function ? colProps(item) : {})}
+              >
+                {expansionMark && (
+                  <span
+                    className={notExpandable.join(" ")}
+                    style={{ fontWeight: "bold", float: "right" }}
+                  >
+                    {isOpened(index) ? "-" : "+"}
+                  </span>
+                )}
+                {cell instanceof Function ? cell(item, query) : item[prop]}
+              </td>
+            )
+          )}
+        </tr>
+      );
+
+      rows.push(<tr key={`${index}-striped-hack`} className={"hidden"} />);
+
+      function revertHidden(hiddenClass) {
+        const all = ["hidden-xs", "hidden-sm", "hidden-md", "hidden-lg"];
+        return all.filter(size => !hiddenClass.includes(size)).join(" ");
+      }
+
+      rows.push(
+        <tr
+          key={`${index}-info`}
+          className={`${notExpandable.join(" ")} ${
+            isOpened(index) ? "" : "hidden"
+          }`}
+        >
+          {withButtons && <td />}
```

Comment:
Nie je withButtons niečo iné? Tabuľky, čo obsahujú nejaké tlačidlá, ich väčšinou majú úplne vpravo (prihlásiť/odhlásiť na skúšku, zapísať na predmet). V CSS `.with-buttons-table` iba mení vertical-align (dávno som zabudol prečo). Tu myslím ide skôr o to, aby to nebolo pod stĺpcom "Moje?".
Možno treba nový prop, čo bude počet preskočených stĺpcov. Alebo to vždy dať pod ten stĺpec, čo má expansionMark.